app/views/govuk_publishing_components/components/_table.html.erb

```diff
@@ -2,46 +2,63 @@
   caption ||= nil
   head ||= []
   rows ||= []
-  first_cell_is_header ||=false
+  first_cell_is_header ||= false
   caption_classes ||= nil
-  sortable ||=false
+  sortable ||= false
+  filterable ||= false
 %>
 
-<%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, caption, {
-    sortable: sortable,
-    caption_classes: caption_classes
-  }) do |t| %>
+<div <% if filterable %> data-module="table-filter" <% end %> >
+  <% if filterable %>
+    <div class="govuk-table-filter">
+      <%= render "govuk_publishing_components/components/input", {
```

Comment:
Something to flag for deployment - because this component now relies on another component, we need to make sure that every application that uses this component also includes the assets for the input component. Although in this case it's fine, because `static` includes the assets for the input component already, so no need to worry. I'm just raising this for awareness.